src/Promise.fs

```diff
@@ -141,3 +141,16 @@ module Promise =
 [<AutoOpen>]
 module PromiseImpl =
     let promise = Promise.PromiseBuilder()
+
+    type Promise.PromiseBuilder with
+        [<CustomOperation("and!", IsLikeZip=true)>]
+        member x.Merge(a, b, [<ProjectionParameter>] resultSelector : _ -> _) =
+            promise {
+                Promise.start a
+                Promise.start b
+                let! a' = a
+                let! b' = b
+                return resultSelector a' b'

```

Comment:
There is likely a more efficient way to make this work with `Promise.Parallel`, but the types _could_ be different, and `Promise.Parallel` wants them to be the same. I'm open to suggestions.